arcgis-ios-sdk-samples/Augmented reality/Navigate in AR/NavigateARNavigatorViewController.swift

```diff
@@ -0,0 +1,286 @@
+    ///
+    /// - Parameters:
+    ///   - route: The new `AGSRoute` to set.
+    ///   - completion: A closure to execute on successfully setting the new route and graphic.
+    func setRouteAndGraphic(route: AGSRoute, onSuccess completion: (() -> Void)? = nil) {
+        currentRoute = route
+        let originalPolyline = route.routeGeometry!
+        elevationSource.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+                self.setStatus(message: "Failed to load elevation source.")
+            } else {
+                self.addElevationToPolyline(polyline: originalPolyline) { [weak self] (newPolyline: AGSPolyline) in
+                    self?.routeGraphic.geometry = newPolyline
+                    completion?()
+                }
+            }
+        }
+    }
+    
+    /// Densify the polyline geometry so the elevation can be adjusted every 0.3 meters,
+    /// and add an elevation to the geometry.
+    ///
+    /// - Parameters:
+    ///   - polyline: The polyline geometry of the route.
+    ///   - z: A `Double` value representing z elevation.
+    ///   - completion: A completion closure to execute after the polyline is generated with success or not.
+    func addElevationToPolyline(polyline: AGSPolyline, elevation z: Double = 3, completion: @escaping (AGSPolyline) -> Void) {
+        if let densifiedPolyline = AGSGeometryEngine.densifyGeometry(polyline, maxSegmentLength: 0.3) as? AGSPolyline {
+            let polylinebuilder = AGSPolylineBuilder(spatialReference: densifiedPolyline.spatialReference)
+            let allPoints = densifiedPolyline.parts.array().flatMap { $0.points.array() }
+            
+            let buildGroup = DispatchGroup()
+            allPoints.forEach { point in
+                buildGroup.enter()
+                elevationSurface.elevation(for: point) { [weak self] (elevation: Double, error: Error?) in
+                    defer { buildGroup.leave() }
+                    if let newPoint = AGSGeometryEngine.geometry(bySettingZ: elevation + z, in: point) as? AGSPoint {
+                        // Put the new point 3 meters above the ground elevation.
+                        polylinebuilder.add(newPoint)
+                    } else if let error = error {
+                        self?.presentAlert(error: error)
+                    }
+                }
+            }
+            
+            buildGroup.notify(queue: .main) {
+                completion(polylinebuilder.toGeometry())
+            }
+        } else {
+            setStatus(message: "Failed to add elevation to route line.")
+            completion(polyline)
+        }
+    }
+    
+    // MARK: Actions
+    
+    @IBAction func startTurnByTurn(_ sender: UIBarButtonItem) {
+        routeTracker = AGSRouteTracker(routeResult: routeResult, routeIndex: 0)
+        if routeTask.routeTaskInfo().supportsRerouting {
```

Comment:
This condition is failing for me and unable to get the turn by turn directions and voice guidance

Reply:
It also fails for me, and going into else still allow my device to provide route status...

I added some prompt to this function, but it might not help too much.

I'll discuss with Nathan and figure it out!